src/modules/gateway/sagas.ts

```diff
@@ -200,37 +200,39 @@ function* handlePollPurchaseStatusRequest(
   const { gateway, id } = purchase
 
   try {
-    switch (gateway) {
-      case NetworkGatewayType.TRANSAK:
-        const { transak: transakConfig } = config
-        const transak = new Transak(transakConfig)
-        let statusHasChanged = false
-
-        while (!statusHasChanged) {
-          const {
-            data: { status, transactionHash, errorMessage }
-          }: OrderResponse = yield call([transak, transak.getOrder], id)
-          const newStatus: PurchaseStatus = yield call(
-            [transak, transak.getPurchaseStatus],
-            status
-          )
-          if (newStatus !== purchase.status) {
-            statusHasChanged = true
-            yield put(
-              setPurchase({
-                ...purchase,
-                status: newStatus,
-                txHash: transactionHash || null,
-                failureReason: errorMessage
-              })
+    if (purchase.status === PurchaseStatus.PENDING) {
```

Comment:
In this cases its better to just 
```
if (something) {
  return
}

somethingElse code
```

Instead of wrapping the whole code on an if